src/Property.php

```diff
@@ -143,6 +143,41 @@ protected function cast($value)
         return new $castTo($value);
     }
 
+    protected function castArray(array $values)
+    {
+        foreach ($values as $value) {
+            if (! is_array($value)) {
+                return $this->cast($values);
+            }
+        }
+
+        $castTo = null;
+        $flatTypes = str_replace('[]', '', $this->types);
+
+        foreach ($flatTypes as $type) {
+            if (! is_subclass_of($type, DataTransferObject::class)) {
+                continue;
+            }
+
+            $castTo = $type;
+
+            break;
+        }
+
+        if (! $castTo) {
+            return $values;
+        }
+
+        $casts = array_map(
```

Comment:
Again I think a loop would be the better choice here, in light of performance.

Reply:
I did some microbenchmarks using this in `\Spatie\DataTransferObject\Property::set`:
```php
            $start = microtime(true);
            for ($i = 0; $i < 1000000; $i++) {
                $this->castArray($value);
            }
            var_dump(microtime(true) - $start);exit(1);
```
I've then switch between this PR and a simply foreach loop:
```php
        $tmp = [];
        foreach ($values as $value) {
            $tmp[] = is_array($values)
                ? new $castTo($value)
                : null;
        }
        return $tmp;
```

With `foreach` (after a few runs, i.e. "hot system"): consistently `3.2s`

With `array_map`: `3.4s`

(Note: this was benchmark before the latest force-push which changed the method slightly)

Reply:
While performance is clearly a non issue, I'm inclined to feel that a loop is easier to read, follow, and make changes to in future; I believe it's the better approach in this instance as opposed to `array_map`.